packages/world/src/systems/DisplaySystem.sol

```diff
@@ -42,19 +45,22 @@ contract DisplaySystem is System {
       machineEnergyLevel = machineData.energy;
     }
     if (machineEnergyLevel > 0) {
-      // We can call the program directly as we are a root system
-      return IDisplayProgram(baseEntityId.getProgramAddress()).getDisplayContent(baseEntityId);
+      bytes memory _getDisplayURI = abi.encodeCall(IHooks.getDisplayURI, (caller, baseEntityId, extraData));
+      bytes memory returnData = baseEntityId.getProgram().staticcallOrRevert(_getDisplayURI);
+      return abi.decode(returnData, (string));
     }
 
-    return DisplayContentData({ contentType: DisplayContentType.None, content: bytes("") });
+    return "";
   }
 
-  function setDisplayContent(EntityId entityId, DisplayContentData memory content) public {
+  function setDisplayURI(EntityId /* caller */, EntityId entityId, string memory uri) public {
```

Comment:
is this needed anymore now that `getDisplayURI` is a getter?

Reply:
Yes, want to sync with dvhani about that, not sure if we still want to support setting display data without a program

Reply:
we discussed at some point that adding displays would increase the forcefield's energy drain rate. Maybe the client could filter specific uri schemas if no forcefield? like only allow static text if no forcefield. But then it wouldn't really be part of the physics

Reply:
yea one thing about displays is that what they show is not really part of the physics, cuz clients can render whatever they want lol